src/main/java/net/snowflake/ingest/streaming/internal/ParameterProvider.java

```diff
@@ -0,0 +1,106 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/** Utility class to provide configurable constants */
+public class ParameterProvider {
+  static final String BUFFER_FLUSH_INTERVAL_IN_MILLIS_MAP_KEY =
+      "STREAMING_INGEST_CLIENT_SDK_BUFFER_FLUSH_INTERVAL_IN_MILLIS";
+  static final String BUFFER_FLUSH_CHECK_INTERVAL_IN_MILLIS_MAP_KEY =
+      "STREAMING_INGEST_CLIENT_SDK_BUFFER_FLUSH_CHECK_INTERVAL_IN_MILLIS";
+  static final String INSERT_THROTTLE_INTERVAL_IN_MILLIS_MAP_KEY =
+      "STREAMING_INGEST_CLIENT_SDK_INSERT_THROTTLE_INTERVAL_IN_MILLIS";
+  static final String INSERT_THROTTLE_THRESHOLD_IN_PERCENTAGE_MAP_KEY =
+      "STREAMING_INGEST_CLIENT_SDK_INSERT_THROTTLE_THRESHOLD_IN_PERCENTAGE";
+
+  // Default values
+  static final long BUFFER_FLUSH_INTERVAL_IN_MILLIS_DEFAULT = 1000;
+  static final long BUFFER_FLUSH_CHECK_INTERVAL_IN_MILLIS_DEFAULT = 100;
+  static final long INSERT_THROTTLE_INTERVAL_IN_MILLIS_DEFAULT = 500;
+  static final long INSERT_THROTTLE_THRESHOLD_IN_PERCENTAGE_DEFAULT = 5;
+
+  /** Map of parameter name to parameter value. This will be set by client/configure API Call. */
+  private Map<String, Object> parameterMap = new HashMap<>();
+
+  /**
+   * Construct empty ParameterProvider that will supply default values until updated by
+   * client/configure API call
+   */
+  public ParameterProvider() {}
+
+  private void updateValue(
+      String key, Object defaultValue, Map<String, Object> parameterOverrides, Properties props) {
+    if (parameterOverrides != null && props != null) {
```

Comment:
props should be always be not null, right? So we should be checking
1. parameterOverrides is null or not
2. parameterOverrides.get(key) is null or not
3. props.get(key) is null or not